### PR TITLE
Optional match

### PR DESCRIPTION
**Pull request by DvirDukhan**

### CODE REVIEW

File: src/resultSet.js [hunk 1]
```diff
@@ -188,13 +191,16 @@ class ResultSet {
...
-     */
+	/**
+	 * Parse a raw edge representation into an Edge object.
+	 * @async
+	 * @param {object[]} cell raw edge representation
+	 * @returns {Edge} Edge object.
+	 */
 	async parseEdge(cell) {
+		// In case of optional match returned a null edge.
+		if (cell.length == 2) return null;
```
Comment (gkorland) [inline src/resultSet.js:202]: @DvirDukhan what if `cell.length <= 1`?

File: test/redisGraphAPITest.js [hunk 2]
```diff
@@ -455,4 +455,44 @@ describe("RedisGraphAPI Test", () => {
 			);
 		}
 	});
+
+	it("testOptionalMatch", async () => {
```
Comment (gkorland) [inline test/redisGraphAPITest.js:459]: @DvirDukhan can you please also add test for "Left outer join"? meaning mixed null and not null values in the same record

Comment (DvirDukhan) [reply test/redisGraphAPITest.js:459]: there is one
`MATCH (a) OPTIONAL MATCH (a)-[e]->(b) RETURN a, e, b ORDER BY ID(a)`

File: src/resultSet.js (hunk 1 above)
Comment (DvirDukhan) [reply src/resultSet.js:202]: fixed

File: test/redisGraphAPITest.js (hunk 2 above)
Comment (gkorland) [reply test/redisGraphAPITest.js:459]: Sorry I missed `assert.notEqual(record.get(0), null);`, why not really check the value?
